### PR TITLE
Fix #2437 Added details to Map

### LINKED ISSUE #2437: Add details section for the map card and burger menu
### Description
The map needs a details section containing an editor for adding video, link, etc.

### Please indicate if this issue is related to a bug or a new feature request
- [x] New Feature

### In case of New Feature (otherwise remove this paragraph)
*Acceptance Criteria - AC*

- [ ] Managing and editing of the details section
- [ ] Add it to the card in the Home
- [ ] Add it to the burger menu in the MapViewer

### Other useful information (optional):
see the mockups [here](http://demo.geo-solutions.it/mockups/mapstore2/#/maps-properties)

### PR DESCRIPTION
**Pull request by MV88**
## Description
Added details in viewer page (under burger menu) if they are present for the opened map.
Added details section in map card
various actions available for details:
- create
- modify
- delete 
- show readonly from the card tools 

## Issues
 - Fix #2437

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

**What is the current behavior?** (You can also link to an open issue here)
Actually is not possible to add details to the map and therefore to change/delete them

**What is the new behavior?**
it is possible from the homepage map card.

**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: 

**Other information**:
You need to add DETAILS category to the configured geostore

### CODE REVIEW

File: web/client/epics/maps.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+            params.resource = {
+                category: "DETAILS",
+                userParams: userParamsSelector(state),
+                metadata: {name: uuidv1()},
+                value,
+                permissions: mapPermissionsFromIdSelector(state, mapId),
+                optionsAttr: {},
+                optionsRes: {}
+            };
+            params.messages = currentMessagesSelector(state);
```
Comment (offtherailz) [inline web/client/epics/maps.js]: Notification system already provide localization. You don't need any access to localization system. Remove all unnecessary code for notifications and use message IDs.

File: web/client/epics/maps.js [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                actions.push(mapDeleted(mapId, "failure", map.error));
+            }
+            if (map.resType === "success") {
+                actions.push(mapDeleted(mapId, "success"));
+                if ( isMapsLastPageSelector(state)) {
+                    actions.push(loadMaps(false, state.maps.searchText || ConfigUtils.getDefaults().initialMapFilter || "*"));
+                }
+            }
+            if (map.resType === "success" && details.resType === "success" && thumbnail.resType === "success") {
+                actions.push(basicSuccess({ message: LocaleUtils.getMessageById(state.locale.messages, "maps.feedback.allResDeleted") + name }));
```
Comment (offtherailz) [inline web/client/epics/maps.js]: message already supports message as msgId

File: web/client/epics/maps.js (hunk 1 above)
Comment (MV88) [reply web/client/epics/maps.js]: Notification system doesn't support custom params just localized strings.
becasue internally they use [getMessageById](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/notifications/NotificationContainer.jsx#L83)
 as you can see i'm adding the name of the map in the notification message which i can simply remove.  suggestions?

Comment (offtherailz) [reply web/client/epics/maps.js]: remove name for now (and all message retrivial code)

**Pull request merged**